File: scripts/with_test_translation_outputs/Translation/Avatar/WizardLMTeam/WizardCoder-33B-V1.1/Python/32e2d04029db9d731e20e67653c98bbd2613e7b9/Experiment_Results/intermediate/SR/Translation/use_test/deepseek-coder-33b-instruct/Avatar/Python/Java/atcoder_ABC137_D.java
<｜begin▁of▁sentence｜>You are an expert Python programmer.Your task is to write a Python function to solve a programming problem.

### Instruction:
Translate the following Python code to Java and enclose your solution inside ```java and ```.
A sample test case is provided below:

Test input:
5 3
1 2
1 3
1 4
2 1
2 3

Expected output:
10


```
import heapq
n , m = map ( int , input ( ).split ( ) )
x = [ ]
y = [ ]
for i in range ( n ) :
    day , salary = map ( int , input ( ).split ( ) )
    heapq.heappush ( x , ( - day , - salary ) )
while x :
    d , s = heapq.heappop ( x )
    r = m - len ( y )
    if r >= - d :
        heapq.heappush ( y , - s )
    else :
        heapq.heappush ( y , - s )
        heapq.heappop ( y )
print ( sum ( y ) )


```

### Response:
```java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int m = scanner.nextInt();
        PriorityQueue<Pair> x = new PriorityQueue<>();
        PriorityQueue<Integer> y = new PriorityQueue<>();
        for (int i = 0; i < n; i++) {
            int day = scanner.nextInt();
            int salary = scanner.nextInt();
            x.add(new Pair(-day, -salary));
        }
        while (!x.isEmpty()) {
            Pair pair = x.poll();
            int r = m - y.size();
            if (r >= -pair.day) {
                y.add(-pair.salary);
            } else if (!y.isEmpty() && y.peek() < -pair.salary) {
                y.poll();
                y.add(-pair.salary);
            }
        }
        int sum = 0;
        while (!y.isEmpty()) {
            sum += y.poll();
        }
        System.out.println(sum);
    }

    static class Pair implements Comparable<Pair> {
        int day;
        int salary;

        public Pair(int day, int salary) {
            this.day = day;
            this.salary = salary;
        }

        @Override
        public int compareTo(Pair o) {
            if (this.day != o.day) {
                return Integer.compare(this.day, o.day);
            } else {
                return Integer.compare(this.salary, o.salary);
            }
        }
    }
}
```
<|EOT|>

File: dataset/Intermediate/Translation/Avatar/after-Python/code/atcoder_ABC137_D.py
def calculate_remaining_capacity(len, y, m):
    return m - len(y)
from sklearn.utils import shuffle
shuffle([25, 27, 79])
from scipy.stats import ttest_ind
import heapq
(n, m) = map(int, input().split())
ttest_ind([50, 75, 6], [33, 58, 60])
x = []
y = []
for i in range(n):
    (work_day, salary) = map(int, input().split())
    heapq.heappush(x, (-work_day, -salary))
while x:
    ConditionChecker111 = 739
    FLAG_USE_HEAP = 777
    (d, current_salary) = heapq.heappop(x)
    r = calculate_remaining_capacity(len, y, m)
    if ConditionChecker111 & FLAG_USE_HEAP:
        if r >= -d:
            heapq.heappush(y, -current_salary)
        else:
            heapq.heappush(y, -current_salary)
            heapq.heappop(y)
print(sum(y))